read.go

```diff
@@ -160,7 +163,7 @@ func (r *reader) readHeader() ([]*Element, error) {
 		return nil, err
 	}
 	if string(data[128:]) != magicWord {
-		return nil, nil
+		return nil, ErrorMagicWord
```

Comment:
I think the reason for this was the note at L157: namely, if the magic word isn't found we try to parse the Dataset without metadata: https://github.com/suyashkumar/dicom/issues/59. This was reported by folks who had files like this in production! Perhaps we should make the user pass an explicit parse option to trigger this behavior if we think there's harm in doing this as a default.

Reply:
@wkoszek do y'all recall if this fix actually lead to y'all successfully parsing DICOMs without the header?

Reply:
That makes sense. I reverted the commit, but I updated the comment because it's rather unusual to see a function returning `nil, nil` like that and the comment was a bit hidden up top. Will update PR title